homeassistant/components/soundtouch/media_player.py

```diff
@@ -437,11 +437,16 @@ def get_zone_info(self):
         # slaves for some reason. To compensate for this shortcoming we have to fetch
         # the zone info from the master when the current device is a slave until this is
         # fixed in the SoundTouch API or libsoundtouch, or of course until somebody has a
-        # better idea on how to fix this
-        if zone_status.is_master:
+        # better idea on how to fix this.
+        # In addition to this shortcoming, libsoundtouch seems to report the "is_master"
+        # property wrong on some slaves, so the only reliable way to detect if the current
+        # devices is the master, is by comparing the master_id of the zone with the device_id
+        if zone_status.master_id is self._device.config.device_id:
             return self._build_zone_info(self.entity_id, zone_status.slaves)
 
-        master_instance = self._get_instance_by_ip(zone_status.master_ip)
+        # The master device has to be searched by it's ID and not IP since libsoundtouch / BOSE API
+        # do not return the IP of the master for some slave objects/responses
+        master_instance = self._get_instance_by_id(zone_status.master_id)
         master_zone_status = master_instance.device.zone_status()
```

Comment:
This call will error if the `_get_instance_by_id` call returns `None`.

Reply:
fixed now